### PR TITLE
Restructure the rendering of the flight plan

### PR DESCRIPTION
**Pull request by pleroy**
This ensures that we don't change state between Layout and Repaint, and it eliminates a funky null pointer exception when first creating a flight plan.

### CODE REVIEW

File: ksp_plugin_adapter/flight_planner.cs [hunk 1]
```diff
@@ -49,175 +54,194 @@ class FlightPlanner : SupervisedWindowRenderer {
...
+        return;
+      }
 
-          FlightPlanAdaptiveStepParameters parameters =
-              plugin_.FlightPlanGetAdaptiveStepParameters(vessel_guid);
+      if (plugin_.FlightPlanExists(vessel_guid)) {
+        RenderFlightPlan(vessel_guid);
+      } else if (UnityEngine.GUILayout.Button("Create flight plan")) {
+        plugin_.FlightPlanCreate(vessel_guid,
+                                  plugin_.CurrentTime() + 1000,
```
Comment (eggrobin) [inline ksp_plugin_adapter/flight_planner.cs]: Bad indent.

File: ksp_plugin_adapter/flight_planner.cs [hunk 2]
```diff
@@ -49,175 +54,194 @@ class FlightPlanner : SupervisedWindowRenderer {
...
+    {
+      string vessel_guid = vessel_?.id.ToString();
+      if (vessel_guid == null ||
+          !plugin_.HasVessel(vessel_guid)) {
+        return;
+      }
 
-          FlightPlanAdaptiveStepParameters parameters =
-              plugin_.FlightPlanGetAdaptiveStepParameters(vessel_guid);
+      if (plugin_.FlightPlanExists(vessel_guid)) {
```
Comment (eggrobin) [inline ksp_plugin_adapter/flight_planner.cs]: Does this not change the set of controls between calls to OnGUI?
the `else if` branch below may change the value of this condition.

File: ksp_plugin_adapter/flight_planner.cs [hunk 3]
```diff
@@ -40,7 +40,12 @@ class FlightPlanner : SupervisedWindowRenderer {
   protected override String Title { get; } = "Flight plan";
 
   protected override void RenderWindow(int window_id) {
-    using (new UnityEngine.GUILayout.VerticalScope()) {
+    // We must ensure that the GUI elements don't change between Layout and
+    // Repaint.  This means that any state change must occur before Layout or
+    // after Repaint.  This if statement implements the former.  It updates the
+    // vessel and the editors to reflect the current state of the plugin and
+    // then proceeds with the UI code.
```
[the remark this answers is not in the document]
Comment (pleroy) [reply ksp_plugin_adapter/flight_planner.cs]: Done.

File: ksp_plugin_adapter/flight_planner.cs (hunk 1 above)
Comment (pleroy) [reply ksp_plugin_adapter/flight_planner.cs]: Done.

File: ksp_plugin_adapter/flight_planner.cs (hunk 2 above)
Comment (pleroy) [reply ksp_plugin_adapter/flight_planner.cs]: Added a comment for the uninitiated.

**Pull request merged**